### PR TITLE
Refactor: Convert PluginPropertiesExtension to Java (#34459)

### PR DESCRIPTION
**Pull request by fabiodrg**
This is the first time I am working at this level with Gradle, so I went through some challenges. One of them is not yet solved, which is **how to access the extra properties from `Project`**. Whenever I try to compile it fails saying the `ext` symbol is not defined (`project.ext`). 
I am looking into Gradle documentation for days, and I can't find a workaround for this. I was having similar issues when accessing properties such `version` or `name`, but I could solve it buy using the methods available. Probably I am misunderstanding how this works. If anyone could provide some help, I would appreciate it a lot.

Due the problem with accessing the extra properties, the following methods are not completed, but they aren't used at the moment, so I decided to create the pull request anyway, and hopefully with the community and elastic team insights I will be able to complete this task.
- `setLicenseFile()`
- `setNoticeFile()`

### CODE REVIEW

**Comment by atorok** [comment]
@fabiodrg666 `project.ext`  in java is `project.getExtensions()                    .getByType(ExtraPropertiesExtension.class)`. Missing properties on a project are looked up from a number of places, extensions are among those,  and `ext` is a special extension of type `ExtraPropertiesExtension`.  I do think the setters are used. note that Groovy calls setters on assignment if they are available.

File: buildSrc/src/main/java/org/elasticsearch/plugin/PluginPropertiesExtension.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A container for plugin properties that will be written to the plugin descriptor, for easy
+ * manipulation in the gradle DSL.
+ */
+public class PluginPropertiesExtension {
+    
+    @Input
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/plugin/PluginPropertiesExtension.java]: In groovy, the `Input` annotation adds the get/set capabilities, and is automatically tracked. However, that is for tasks. I'm not sure it actually makes sense (and was erroneous but harmless in the old groovy code). @atorok can you confirm my understanding? If so, I think all these Input annotations should be removed.

Comment (fabiodrg666) [reply buildSrc/src/main/java/org/elasticsearch/plugin/PluginPropertiesExtension.java]: Thanks for your feedback, I will remove the `Input` annotations.

**Comment by danielmitterdorfer** [comment]
@fabiodrg666 are you still interested in getting this PR in?

**Comment by atorok** [comment]
Overlaps with #39605

**Comment by atorok** [comment]
@fabiodrg666 I'm going to close this, please reopen if you plan to work on it

**Pull request closed without merging**